checker/jdk/nullness/src/org/w3c/dom/Text.java

```diff
@@ -0,0 +1,190 @@
+ * an attribute.
+ * <p>See also the <a href='http://www.w3.org/TR/2004/REC-DOM-Level-3-Core-20040407'>Document Object Model (DOM) Level 3 Core Specification</a>.
+ */
+public interface Text extends CharacterData {
+    /**
+     * Breaks this node into two nodes at the specified <code>offset</code>,
+     * keeping both in the tree as siblings. After being split, this node
+     * will contain all the content up to the <code>offset</code> point. A
+     * new node of the same type, which contains all the content at and
+     * after the <code>offset</code> point, is returned. If the original
+     * node had a parent node, the new node is inserted as the next sibling
+     * of the original node. When the <code>offset</code> is equal to the
+     * length of this node, the new node has no data.
+     * @param offset The 16-bit unit offset at which to split, starting from
+     *   <code>0</code>.
+     * @return The new node, of the same type as this node.
+     * @exception DOMException
+     *   INDEX_SIZE_ERR: Raised if the specified offset is negative or greater
+     *   than the number of 16-bit units in <code>data</code>.
+     *   <br>NO_MODIFICATION_ALLOWED_ERR: Raised if this node is readonly.
+     */
+    public @Nullable Text splitText(int offset)
+                          throws DOMException;
+
+    /**
+     * Returns whether this text node contains <a href='http://www.w3.org/TR/2004/REC-xml-infoset-20040204#infoitem.character'>
+     * element content whitespace</a>, often abusively called "ignorable whitespace". The text node is
+     * determined to contain whitespace in element content during the load
+     * of the document or if validation occurs while using
+     * <code>Document.normalizeDocument()</code>.
+     * @since DOM Level 3
+     */
+    @Pure public boolean isElementContentWhitespace();
+
+    /**
+     * Returns all text of <code>Text</code> nodes logically-adjacent text
+     * nodes to this node, concatenated in document order.
+     * <br>For instance, in the example below <code>wholeText</code> on the
+     * <code>Text</code> node that contains "bar" returns "barfoo", while on
+     * the <code>Text</code> node that contains "foo" it returns "barfoo".
+     *
+     * <pre>
+     *                     +-----+
+     *                     | &lt;p&gt; |
+     *                     +-----+
+     *                       /\
+     *                      /  \
+     *               /-----\    +-------+
+     *               | bar |    | &amp;ent; |
+     *               \-----/    +-------+
+     *                              |
+     *                              |
+     *                           /-----\
+     *                           | foo |
+     *                           \-----/
+     * </pre>
+     * <em>Figure: barTextNode.wholeText value is "barfoo"</em>
+     *
+     * @since DOM Level 3
+     */
+    @Pure public @Nullable String getWholeText();
```

Comment:
The node value and DOMString can be null. Text class is derived from parent Node class.